ocaml/testsuite/tests/typing-layouts-float64/unboxed_floats_alpha.ml

```diff
@@ -0,0 +1,118 @@
+(* TEST
+   flags = "-extension layouts_alpha"
+*)
+
+(* This file contains various tests for float#.  It's not an expect test to make
```

Comment:
Yeah I think given the code for dealing with C call exttypes has been altered, we should add these tests now.

Reply:
I've [pushed a commit](https://github.com/ocaml-flambda/flambda-backend/pull/1528/commits/9e89ec9d6e93be21d7d33c67c1f59d61310c2e48) that tests various `external` stuff.

I didn't actually allow your example, though.  `[@@unboxed]` means to unbox all the types in the declaration, and `float#` can't be unboxed.  To achieve what you wanted, the user should write:
```
external sin : (float[@unboxed]) -> float# = "sin"
```
Of course, we could allow the other thing, but it seems good to make users be clear around C interfaces, which are easy to get wrong (I wrote many bugs in preparing these tests).  Also, this was less work.

I added checks and errors for a couple other things, too.  In particular:
- When writing a C stub involving a non-value layout, you must supply separate bytecode and native code versions (because the bytecode version will be boxed).  If we allowed just one version, the generated code would just segfault (or worse) - I tried.
- We don't allow the legacy unboxed float declaration style (`external foo : ... = "foo" "bar" "float"`) when non-value layouts are involved.

Reply:
Looks good. I've reviewed the commit; no comments. Thanks.